tests/test_authenticator.py

```diff
@@ -958,7 +966,7 @@ def test_authenticated_patron_basic(self):
         patrondata = PatronData(
             permanent_id=patron.external_identifier,
             authorization_identifier=patron.authorization_identifier,
-            username=patron.username
+            username=patron.username, neighborhood="Achewood"
```

Comment:
Just to make sure, right here the `patron` already exists and is found, so what's happening is this value is added to the `patron` object in `get_or_create_patron` (this is similar to the test above). From here until the test in line 992, does the `PatronData.authenticate` method get called? That's the other place where `patron` is assigned the neighborhood value.

Reply:
Yes, in this case the patron is looked up by get_or_create_patron, and at that point Patron.neighborhood is set. 

`apply_patrondata` is called when the patron is found through `local_patron_lookup`, which can't use `get_or_create_patron` because the query is too complicated. (`local_patron_lookup` is actually the most common way a Patron object is created in the real world.)